C2GUILauncher/src/ModDownloader.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C2GUILauncher
+{
+    class ModDownloader
+    {
+        const string InjectorDllName = "XAPOFX1_5.dll";
+
+        const string ModCachePath = ".mod_cache";
+        const string PluginPath = "TBL\\Binaries\\Win64\\Plugins";
+        const string GithubBaseURL = "https://github.com";
+
+
+        public static IEnumerable<DownloadTask> DownloadModFiles(bool debug)
+        {
+            // All Chiv2-Community dll releases have an optional _dbg suffix for debug builds.
+            var downloadFileSuffix = debug ? "_dbg.dll" : ".dll";
+
+            // These are the core mods necessary for asset loading, server hosting, server browser usage, and the injector itself.
+            var coreMods = new List<DownloadTarget>() {
+                new DownloadTarget($"{GithubBaseURL}/Chiv2-Community/C2PluginLoader/releases/latest/download/XAPOFX1_5{downloadFileSuffix}", Path.Combine(ModCachePath, InjectorDllName)),
+                new DownloadTarget($"{GithubBaseURL}/Chiv2-Community/C2AssetLoaderPlugin/releases/latest/download/C2AssetLoaderPlugin{downloadFileSuffix}", Path.Combine(PluginPath, "C2AssetLoaderPlugin.dll")),
+                new DownloadTarget($"{GithubBaseURL}/Chiv2-Community/C2ServerPlugin/releases/latest/download/C2ServerPlugin{downloadFileSuffix}", Path.Combine(PluginPath, "C2ServerPlugin.dll")),
+                new DownloadTarget($"{GithubBaseURL}/Chiv2-Community/C2BrowserPlugin/releases/latest/download/C2BrowserPlugin{downloadFileSuffix}", Path.Combine(PluginPath, "C2BrowserPlugin.dll"))
```

Comment:
This is where we currently set the core mods download list. The paths used could probably be moved in to another module since your new stuff will want to know all of them for injection